src/cider/nrepl/middleware/macroexpand.clj

```diff
@@ -113,10 +113,15 @@
   context of the given :ns, using the provided :expander and :display-namespaces
   options."
   [{:keys [code expander ns] :as msg}]
-  (->> (let [expander-fn (resolve-expander-clj expander)]
-         (binding [*ns* (find-ns ns)]
-           (expander-fn (read-string code))))
-       (walk/prewalk (post-expansion-walker-clj msg))))
+  (with-bindings {clojure.lang.Compiler/LOADER
+                  (if (instance? clojure.lang.Var$Unbound
+                                 @clojure.lang.Compiler/LOADER)
+                    (clojure.lang.DynamicClassLoader. (clojure.lang.RT/baseLoader))
+                    @clojure.lang.Compiler/LOADER)}
```

Comment:
This is the fix

Looks pretty boring but I could use a second pair of 👀s here. @plexus perhaps?

I'd assume this is a reasonable way to instantiate a DCL. I'd guess that it's not a leaky pattern?

Reply:
Looks quite ugly, though, even if it's pretty simple. You might also add a note that's something specific to tools.deps, as I doubt many people will figure out why we need such gymnastics.

Reply:
Definitely can add a comment.

Not 100% sure if exclusive to t.deps though; in the end that should be one of the rawest ways of running Clojure. It can also be framed as 'Lein was doing something for us that we implicitly relied on'.

Draft:

```
Bind LOADER, which can be unbound under certain code paths, particularly when using tools.deps.
```

Reply:
Sounds reasonable to me.

Reply:
Amended with the comment